pkg/readiness/config/health_status_provider.go

```diff
@@ -0,0 +1,37 @@
+package config
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/readiness/health"
+)
+
+// HealthStatusProvider is an interface for providing the 'health.Status' object to the readiness probe. Abstracting out
+// this to the interface allows to perform unit-testing easier as allows to modify the config on-the-fly.
+type HealthStatusProvider interface {
+	HealthStatus() (health.Status, error)
+}
```

Comment:
I think this interface may be an unnecessary level of abstraction. We can already utilize the native `io.Reader` interface to achieve these goals E.g.

```golang
func HealthStatus(reader io.Reader) (health.Status, error) {
	var health health.Status
	fd, err := os.Open(f.HealthStatusFilePath)
	if err != nil {
		return health, err
	}
	defer fd.Close()

	data, err := reader.Read(fd)
	if err != nil {
		return health, err
	}

	err = json.Unmarshal(data, &health)
	return health, err
}
```

We can provide a mock implementation by simply marshalling an instance of health.Status and returning the byte array.

```golang

mockReader := mockReader(healthStatus)
bytes, err := mockReader.Read()
var health health.Status
err = json.Unmarshal(data, &health)
```